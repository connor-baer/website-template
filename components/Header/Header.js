import React from 'react';
import styled from '@emotion/styled';
import { css } from '@emotion/core';

import { getSrc } from '../Image/ImageService';
import {
  fullWidthStyles,
  gridStyles,
  pageWidthStyles
} from '../../styles/shared';

const wrapperStyles = ({ theme }) => css`
  ${theme.mq.kilo} {
    padding-right: 2rem;
    padding-left: 2rem;
  }
`;

const Wrapper = styled('header')(wrapperStyles, fullWidthStyles);

const backgroundStyles = ({ theme, image = {} }) => css`
  width: 100%;
  height: 25rem;
  background-color: ${theme.colors.offBlack};
  background-image: url("${getSrc(image.src, { width: 1500, height: 500 })}");
  background-size: cover;
  background-repeat: no-repeat;
  background-position: center;
  margin-bottom: 3rem;
  position: relative;

  &::after {
    content: '';
    display: block;
    position: absolute;
    width: 100%;
    height: 100%;
    background-color: ${theme.colors.offBlack};
    top: 0;
    opacity: 0.2;
  }

  ${theme.mq.kilo} {
    border-radius: ${theme.borderRadius.giga};
    overflow: hidden;
  }
`;

const Background = styled('div')(backgroundStyles);

const titleStyles = ({ theme }) => css`
  font-size: 3rem;
  color: #fff;
  grid-column: 1 / -1;
  position: absolute;
  bottom: 3rem;
  margin: 0;
  z-index: 1;
  text-shadow: 1px 2px 6px rgba(0, 0, 0, 0.2);

  ${theme.mq.kilo} {
    grid-column: 1 / 10;
    bottom: 4rem;
  }

  ${theme.mq.mega} {
    grid-column: 1 / 8;
    bottom: 5rem;
  }
`;

const Title = styled('h1')(titleStyles);

const contentStyles = () => css`
  position: relative;
  height: 100%;
`;

const Content = styled('div')(gridStyles, pageWidthStyles, contentStyles);

export default function Header({ title, ...rest }) {
  return (
    <Wrapper>
      <Background {...rest}>
        <Content>
          <Title>{title}</Title>
        </Content>
      </Background>
    </Wrapper>
  );
}
