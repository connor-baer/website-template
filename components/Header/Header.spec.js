import React from 'react';

import Header from './Header';

describe('Header', () => {
  function renderHeader(fn, props = {}) {
    return fn(<Header {...props} />);
  }

  const defaultProps = {
    title: 'Pineapple',
    image: {
      src: 'https://source.unsplash.com/pineapple.png'
    }
  };

  it('should render with default styles', () => {
    const { container } = renderHeader(render, defaultProps);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should meet basic accessibility guidelines', async () => {
    const html = renderHeader(renderToHtml, defaultProps);
    const result = await axe(html);
    expect(result).toHaveNoViolations();
  });
});
