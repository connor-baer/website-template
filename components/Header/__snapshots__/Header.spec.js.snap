// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render with default styles 1`] = `
.circuit-6 {
  max-width: 80rem;
  margin-right: auto;
  margin-left: auto;
}

@media (min-width:600px) {
  .circuit-6 {
    padding-right: 2rem;
    padding-left: 2rem;
  }
}

@media (min-width:600px) {
  .circuit-6 {
    padding-right: 2rem;
    padding-left: 2rem;
  }
}

.circuit-4 {
  width: 100%;
  height: 25rem;
  background-color: #15191d;
  background-image: url("https://source.unsplash.com/pineapple.png?w=1500&h=500");
  background-size: cover;
  background-repeat: no-repeat;
  background-position: center;
  margin-bottom: 3rem;
  position: relative;
}

.circuit-4::after {
  content: '';
  display: block;
  position: absolute;
  width: 100%;
  height: 100%;
  background-color: #15191d;
  top: 0;
  opacity: 0.2;
}

@media (min-width:600px) {
  .circuit-4 {
    border-radius: 6px;
    overflow: hidden;
  }
}

.circuit-2 {
  display: grid;
  grid-template-columns: repeat(12,[col-start] minmax(0,1fr));
  grid-column-gap: 1rem;
  max-width: 67.5rem;
  margin-right: auto;
  margin-left: auto;
  padding-right: 1rem;
  padding-left: 1rem;
  position: relative;
  height: 100%;
}

@media (min-width:600px) {
  .circuit-2 {
    padding-right: 2rem;
    padding-left: 2rem;
  }
}

.circuit-0 {
  font-size: 3rem;
  color: #fff;
  grid-column: 1 / -1;
  position: absolute;
  bottom: 3rem;
  margin: 0;
  z-index: 1;
  text-shadow: 1px 2px 6px rgba(0,0,0,0.2);
}

@media (min-width:600px) {
  .circuit-0 {
    grid-column: 1 / 10;
    bottom: 4rem;
  }
}

@media (min-width:900px) {
  .circuit-0 {
    grid-column: 1 / 8;
    bottom: 5rem;
  }
}

<header
  class="circuit-6 circuit-7"
>
  <div
    class="circuit-4 circuit-5"
  >
    <div
      class="circuit-2 circuit-3"
    >
      <h1
        class="circuit-0 circuit-1"
      >
        Pineapple
      </h1>
    </div>
  </div>
</header>
`;
